pkg/remoteresolution/resolver/git/resolver.go

```diff
@@ -94,62 +99,46 @@ func (r *Resolver) GetSelector(_ context.Context) map[string]string {
 	}
 }
 
-// ValidateParams returns an error if the given parameter map is not
+// GetConfigName returns the name of the git resolver's configmap.
+func (r *Resolver) GetConfigName(_ context.Context) string {
+	return git.ConfigMapName
+}
+
+// Validate returns an error if the given parameter map is not
 // valid for a resource request targeting the gitresolver.
 func (r *Resolver) Validate(ctx context.Context, req *v1beta1.ResolutionRequestSpec) error {
-	if len(req.Params) > 0 {
-		return git.ValidateParams(ctx, req.Params)
-	}
-	// Remove this error once validate url has been implemented.
-	return errors.New("cannot validate request. the Validate method has not been implemented.")
+	return git.ValidateParams(ctx, req.Params)
 }
 
-// Resolve performs the work of fetching a file from git given a map of
-// parameters.
-func (r *Resolver) Resolve(ctx context.Context, req *v1beta1.ResolutionRequestSpec) (resolutionframework.ResolvedResource, error) {
-	if len(req.Params) > 0 {
-		origParams := req.Params
-
-		if git.IsDisabled(ctx) {
-			return nil, errors.New(disabledError)
-		}
-
-		params, err := git.PopulateDefaultParams(ctx, origParams)
-		if err != nil {
-			return nil, err
-		}
-
-		g := &git.GitResolver{
-			KubeClient: r.kubeClient,
-			Logger:     r.logger,
-			Cache:      r.cache,
-			TTL:        r.ttl,
-			Params:     params,
+// IsImmutable implements ImmutabilityChecker.IsImmutable
+// Returns true if the revision parameter is a commit SHA (40-character hex string)
```

Comment:
Quick note : as of today, a full git SHA is SHA-1, but tomorrow (git v3), it would/could but SHA-256, so we could try to take this case into account today already ? (or create a follow-up issue)